src/lightly_train/_commands/train_task_helpers.py

```diff
@@ -222,14 +224,70 @@ def get_val_transform(
     return train_model_cls.val_transform_cls(transform_args=val_transform_args)
 
 
+def get_dataset_mmap_filenames(
+    fabric: Fabric,
+    filenames: Iterable[str],
+    mmap_filepath: Path,
+) -> MemoryMappedSequence[str]:
+    """Returns memory-mapped filenames shared across all ranks.
+
+    Filenames are written to mmap_filepath by rank zero and read by all ranks.
+    """
+    if Env.LIGHTLY_TRAIN_MMAP_REUSE_FILE.value and mmap_filepath.exists():
+        # If the file already exists and we are allowed to reuse it, return it.
+        logger.warning(f"Reusing existing memory-mapped file '{mmap_filepath}'.")
+        return memory_mapped_sequence.memory_mapped_sequence_from_file(
+            mmap_filepath=mmap_filepath
+        )
+
+    with fabric.rank_zero_first():
+        memory_mapped_sequence.write_filenames_to_file(
+            filenames=filenames,
+            mmap_filepath=mmap_filepath,
+        )
```

Comment:
```suggestion
        if (fabric.global_rank == 0) or (is_shared_filesystem and fabric.local_rank == 0):
          memory_mapped_sequence.write_filenames_to_file(
              filenames=filenames,
              mmap_filepath=mmap_filepath,
          )
```

You also have to move the `is_shared_filesystem` check above the `write_filenames_to_file` section

Reply:
Should be `not is_shared_filesystem`